From b04becf510da6406eb2671183e334b96f5490c59 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 00:59:31 -0400
Subject: [acpi] dock: Drop the hp_lock mutex from struct dock_station

Message-id: <20140710005931.22146.9138.stgit@gir.stowe>
Patchwork-id: 85551
O-Subject: [RHEL7.1 PATCH 012/105] ACPI / dock: Drop the hp_lock mutex from struct dock_station
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: d460acebd7959cc91e7edc594d90adb9b72a0b05

commit d460acebd7959cc91e7edc594d90adb9b72a0b05
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Sun Jun 30 23:42:51 2013 +0200

    ACPI / dock: Drop the hp_lock mutex from struct dock_station

    The only existing user of the hp_lock mutex in struct dock_station,
    hotplug_dock_devices(), is always called under acpi_scan_lock and
    cannot race with another instance of itself, so drop the mutex
    which is not necessary.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/dock.c b/drivers/acpi/dock.c
index 810d1d7..c107615 100644
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -63,7 +63,6 @@ struct dock_station {
  acpi_handle handle;
  unsigned long last_dock_time;
  u32 flags;
- struct mutex hp_lock;
  struct list_head dependent_devices;
 
  struct list_head sibling;
@@ -351,8 +350,6 @@ static void hotplug_dock_devices(struct dock_station *ds, u32 event)
 {
  struct dock_dependent_device *dd;
 
- mutex_lock(&ds->hp_lock);
-
  /*
   * First call driver specific hotplug functions
   */
@@ -371,7 +368,6 @@ static void hotplug_dock_devices(struct dock_station *ds, u32 event)
   else
    dock_create_acpi_device(dd->handle);
  }
- mutex_unlock(&ds->hp_lock);
 }
 
 static void dock_event(struct dock_station *ds, u32 event, int num)
@@ -893,7 +889,6 @@ static int __init dock_add(acpi_handle handle)
  dock_station->dock_device = dd;
  dock_station->last_dock_time = jiffies - HZ;
 
- mutex_init(&dock_station->hp_lock);
  INIT_LIST_HEAD(&dock_station->sibling);
  INIT_LIST_HEAD(&dock_station->dependent_devices);
 
-- 
1.7.1